tracing/errors.go

```diff
@@ -1,13 +1,24 @@
 package tracing
 
 import (
-	"errors"
 	"fmt"
 )
 
 // errNotSpanHook is the error logged when you pass an object to span.AddHooks
 // that does not conform to any of the span hook interfaces.
-var errNotSpanHook = errors.New("attempting to add non-SpanHook object into span's hook registry")
+type errNotSpanHook struct {
+	obj interface{}
+}
+
+var _ error = (*errNotSpanHook)(nil)
+
+func (e *errNotSpanHook) Error() string {
+	return fmt.Sprintf(
+		"attempting to add non-SpanHook object <%T> into span's hook registry\n%#v",
+		e.obj,
+		e.obj,
+	)
+}
```

Comment:
This is overkill since we don't export it, just use `fmt.Errorf` in place (with `LogError` inside `span.go`) is enough. Also don't use `"\n"` inside error message, `": "` should be enough.